diffrax/integrate.py

```diff
@@ -283,7 +283,7 @@ def _body_fun(_state, _inplace):
                         jnp.where(_pred, __saveat_y, __ys[_save_index])
                     ),
                     _ys,
-                    _saveat_y,
+                    saveat.func(_ts[_save_index], _saveat_y, args)
```

Comment:
Nit: `_ts[save_index]` is set to `jnp.where(_pred, _saveat_y, __ys[_save_index])` above, so we can just pull that our as an intermediate value and access it directly, rather than re-indexing.

Reply:
You're also missing a couple of other spots in this file that need a `func` wrapper: the `saveat.steps` branch, and the `saveat.t0` and `saveat.t1` branch.

Reply:
- Not sure what you mean by the intermediate value bit
- Done